File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cs
public class Solution {
    // LeetCode 271
    public string Encode(IList<string> strs) {
        StringBuilder s = new StringBuilder();
        foreach(string st in strs){
            s.Append(st.Length).Append('#').Append(st);
        }
        return s.ToString();
    }

    public List<string> Decode(string s) {
        int n = s.Length;
        int i = 0;
        List<string> arr = new List<string>();
        while(i<n){
            int j = i;
            while(s[j] != '#') j++;
            int length = int.Parse(s.Substring(i,j-i));
            j++;
            arr.Add(s.Substring(j,length));
            i = j+length;
        }
        return arr;
   }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-2.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
       StringBuilder s = new StringBuilder();
        s.append(strs[0]);
        for(int i = 1;i<strs.length;i++){
            int j = 0;
            while(j<s.length() && j<strs[i].length() && s.charAt(j) == strs[i].charAt(j))
                j++;
            s.setLength(j);
            if(s.length() == 0) return "";
        }
        return s.toString();
    }
}